File: elixir/exam2_challenge.exs
defmodule Exam2 do
  def start do
    name = IO.gets "What is the input string? "
    name = String.replace_trailing(name, "\n", "")

    string_length = String.length(name)
    if string_length == 0 do
      IO.puts "please input string."
      Exam2.start
    else
      IO.puts "#{name} has #{String.length(name)} characters."
    end
  end
end

Exam2.start

File: go/exam2_challenge.go
package main

import "fmt"

func main() {
	var name string

	for {
		fmt.Print("What is the input string? ")
		fmt.Scanf("%s", &name)

		size := len(name)
		if size == 0 {
			fmt.Println("please input string.")
			continue
		}

		fmt.Printf("%s has %d characters.\r\n", name, size)
	}

}
